### ISSUE TITLE
Set data to state making browser stuck (not responding)

### ISSUE THREAD

**Issue by ardyfeb** [ckeditor/ckeditor5-react#37]
Im new on ckeditor, i follow the documentation ckeditor for React
Here my code

 `<Editor onChange={(ev, ed) => this.setState({ content: ed.getData()})}/>`

But when i start typing on ckeditor section, my browser not responding and the content not showing on the textarea

When i try to alert() the data is working

**Comment by pomek** [comment]
In your code, I don't see the [`editor` property](https://github.com/ckeditor/ckeditor5-react#component-properties) which is required to properly work the component.

I checked out your case with my example component:

```jsx
class App extends Component {
	constructor( props ) {
		super( props );

		this.state = {
			content: '<p>Initial data.</p>'
		};
	}

	handleEditorChange() {
		return ( event, editor ) => {
			this.setState( { content: editor.getData() } );
			console.log( this.state );
		}
	}

	render() {
		return (
			<div className="App">
				<CKEditor
					config={{
						plugins: [ Essentials, Heading, Paragraph, Bold, Italic ],
						toolbar: [ 'Heading', '|', 'Bold', 'Italic', 'Undo', 'Redo' ],
					}}
					editor={ClassicEditor}
					data={this.state.content || ''}
					onChange={this.handleEditorChange()}
				/>
			</div>
		);
	}
}
```

and everything works. The state is updated correctly and nothing freezes in the browser.

But if you think that something is broken in the component, describe steps to reproduce the issue and we will be able to check this.

**Comment by ardyfeb** [comment]
Here my full code

```
import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Bookmark from '@material-ui/icons/Bookmark';
import Person from '@material-ui/icons/Person';
import Memory from '@material-ui/icons/Memory';
import Time from '@material-ui/icons/AccessTime';
import MenuItem from '@material-ui/core/MenuItem';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import Paper from '@material-ui/core/Paper';

const Editor = (props) => {
   if (typeof window != 'undefined') {
       const CKEditor = require('@ckeditor/ckeditor5-react')
       const ClassicEditor = require('@ckeditor/ckeditor5-build-classic')
       
       return (
           <CKEditor editor={ClassicEditor} {...props}/>
       )
   } else {
       return ''
   } 
}

class CreatePostView extends Component { 
    state = {
        content: ''
    }
       
    render()  {
        const { classes } = this.props;
                  
        return (
            <div className={classes.root}>
               <Typography className={classes.headline} variant="display1">Buat post baru</Typography>
               <Typography className={classes.subHeadline} variant="body1" gutterBottom>Share your post, it's free</Typography>
               <Grid container spacing={16}>
                  <Grid item xs={12} md={8}>
                     <TextField fullWidth
                                label="Judul"
                                placeholder="Sebuah judul"
                                InputLabelProps={{
                                    shrink: true                                    
                                }}
                                className={classes.input}
                     />
                     <TextField fullWidth
                                select
                                label="Kategori"
                                className={classes.input}
                                value="wkkwwland"
                                
                     >
                        <MenuItem key="1" value="wkkwwland">wkkwland</MenuItem>
                        <MenuItem key="2" value="djdjdj">djdjdjdj</MenuItem>
                     </TextField>
                     <TextField fullWidth
                                label="Tag"
                                placeholder="Pisahkan dengan koma"
                                InputLabelProps={{
                                    shrink: true                                    
                                }}
                                className={classes.input}
                     />
                     <Editor onChange={(ev, ed) => this.setState({ content: ed.getData()})}/>
                  </Grid>
                  <Grid item xs={12} md={4}>
                     <Paper className={classes.sidebarItem}>
                        <Typography variant="title" className={classes.sidebarItemTitle}>Post detail</Typography>
                        <List>
                           <ListItem>
                              <ListItemIcon>
                                 <Bookmark/>
                              </ListItemIcon>
                              <ListItemText primary={
                                                 <Typography variant="subheading">
                                                     Status : <div className={classes.postDetailValue}>{this.props.post_details.status}</div>
                                                 </Typography>
                                             }
                              />
                           </ListItem>  
                           <ListItem>
                              <ListItemIcon>
                                 <Time/>
                              </ListItemIcon>
                              <ListItemText primary={
                                                 <Typography variant="subheading">
                                                     Published : <div className={classes.postDetailValue}>{this.props.post_details.published_time ? this.props.post_details.published_time : new Date().toLocaleString()}</div>
                                                 </Typography>
                                             }
                              />
                           </ListItem> 
                           <ListItem>
                              <ListItemIcon>
                                 <Memory/>
                              </ListItemIcon>
                              <ListItemText primary={
                                                 <Typography variant="subheading">
                                                     Autosave : <div className={classes.postDetailValue}>{this.props.post_details.autosave ? 'ON' : 'OFF'}</div>
                                                 </Typography>
                                             }
                              />
                           </ListItem> 
                           <ListItem>
                              <ListItemIcon>
                                 <Person/>
                              </ListItemIcon>
                              <ListItemText primary={
                                                 <Typography variant="subheading">
                                                     Author : <div className={classes.postDetailValue}>{this.props.post_details.author}</div>
                                                 </Typography>
                                             }
                              />
                           </ListItem>                         
                        </List>
                        <div className={classes.sidebarFooterAction}>
                            <Grid container>
                                <Grid item xs={6}>
                                    <Button variant="contained" color="primary">Simpan draft</Button>
                                </Grid>
                                <Grid container item xs={6} direction="column" alignItems="flex-end">
                                    <Button variant="contained" color="primary">Publish</Button>
                                </Grid>
                            </Grid>
                        </div>
                     </Paper> 
                     <Paper className={classes.sidebarItem}>
                        <Typography variant="title" className={classes.sidebarItemTitle}>Other meta</Typography>
                        <div className={classes.otherMetaForm}>
                            <div onClick={this.props.pickFeaturedImage} className={classes.featuredImage} id="featured_image">
                                <Typography id="featured_image_text" variant="title" className={classes.featuredImageText}>Featured image</Typography>
                                <input onChange={this.props.handleFeaturedImage} type="file" id="featured_image_picker" accept="image/*" style={{ display: 'none'}}/>
                            </div>
                            <TextField fullWidth
                                label="Slug"
                                placeholder="Kosongkan jika ingin default"
                                InputLabelProps={{
                                    shrink: true                                    
                                }}
                                className={classes.input}
                             />
                            <TextField id="excerpt" 
                                       label="Deskripsi"
                                       multiline 
                                       fullWidth
                                       variant="outlined"
                                       rows={4}
                                       className={classes.input}
                                       
                             />
                        </div>
                     </Paper>
                  </Grid>                     
               </Grid>              
            </div>
        )
    }
}

const styles = theme => ({
    root: {
        margin: theme.spacing.unit,
        [theme.breakpoints.up('md')]: {
            margin: 20
        }
    },
    headline: {
        fontWeight: 'bold',
        marginTop: 20,
    },
    subHeadline: {
        marginBottom: 30,
        paddingLeft: 2
    },
    featuredImage: {
        marginTop: 20,
        borderRadius: 6,
        height: 150,
        border: 'solid 2px #6200EE',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundSize: '100% 100%',
        marginBottom: 20,
        textAlign: 'center',
    },
    featuredImageText: {
        marginTop: 60               
    },
    input: {
        marginBottom: 20 ,
    },
    sidebarItem: {
        marginBottom: theme.spacing.unit * 2
    },
    sidebarItemTitle: {
        padding: 20,
        paddingBottom: 0
    },
    sidebarFooterAction: {
        padding: 20,
        paddingTop: 0
    },
    otherMetaForm: {
        padding: 20,
        paddingTop: 0,
        paddingBottom: 0
    },
    postDetailValue: {
        textDecoration: 'underline',
        fontWeight: 'bold',
        display: 'inline'
    }
})

export default withStyles(styles)(CreatePostView)
```

**Comment by ardyfeb** [comment]
Solved ! I forgot to set data props. Thanks, closed

**Comment by ma2ciek** [comment]
That makes sense now as we check if the state of the editor is the same as the data which comes from outside to not create an infinite update loop - see these lines: https://github.com/ckeditor/ckeditor5-react/blob/383b25d35beb54788f91185ea6c30615b56bb2b6/src/ckeditor.jsx#L18-L22

**Comment by ardyfeb** [comment]
And i got new problem again, the underline button not showing up

**Comment by pomek** [comment]
Could you show us the `Editor` (`const Editor = (props) => { ...`) function? Does any error occur in the console?

**Comment by ardyfeb** [comment]
When i test my web app on my phone,

The ckeditor overriding parent width
![screenshot_chrome_beta_20180919-154455](https://user-images.githubusercontent.com/30074366/45742062-4b6b4a80-bc23-11e8-88a7-6012bb8eb5f7.png)

When set the toolbar just 'bold, italic, underline, heading'
![screenshot_chrome_beta_20180919-154552](https://user-images.githubusercontent.com/30074366/45742126-781f6200-bc23-11e8-9fc9-f1ed3f69e5f0.png)
Everything is oke

But the underline button not showing up
Here my config
```
const editorConfig = {
    toolbar: [
        'heading', '|', 'bold', 'italic', 'underline'
    ]
}
```

**Comment by pomek** [comment]
The [`Underline`](https://github.com/ckeditor/ckeditor5-basic-styles/blob/0b1aebd617e60fe680f5093d185dbcaf15f06d3f/src/underline.js) plugin isn't added to [ckeditor5-build-classic](https://github.com/ckeditor/ckeditor5-build-classic/blob/67c3241b3628239f909e33d9b433e891e51c7216/src/ckeditor.js).

If you want to use this plugin, you must customize the build – https://github.com/ckeditor/ckeditor5-react#customizing-the-builds.

**Comment by ma2ciek** [comment]
Hi, @ardyfeb,

If you have more issues (like this with the editor width) please report it as a separate ticket in the correct repository. It has nothing to do with the `ckeditor5-react`, nor the title of this issue. It's an easy way to create a mess, thus it can make this issue not helpful for others.

**Comment by developingAlex** [comment]
I experienced this behaviour recently which was very frustrating, as a controlled component the ckeditor's editor would not allow you to edit it smoothly, it would be very slow or lock up, or sometimes run in an infinite loop that freezed up the browser. 
In the console would be printed useless errors pointing to a line in ckeditors minified javascript and a message to the effect of "…t is undefined…"

What fixed it for me was to delete my node_modules folder and reinstall all packages again with `npm install`.

Posting this in case it helps anyone else as this was one of the threads I kept coming back to in trying to debug this.
